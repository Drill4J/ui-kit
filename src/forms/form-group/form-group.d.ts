/// <reference types="react" />
interface Props {
    className?: string;
    children?: React.ReactNode;
    label?: string;
    optional?: boolean;
    actions?: React.ReactNode;
}
export declare const FormGroup: {
    ({ className, children, label, optional, actions, }: Props): JSX.Element;
    displayName: string;
    __docgenInfo: {
        description: string;
        displayName: string;
        props: {
            className: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            label: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            optional: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            actions: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
        };
    };
};
export {};
