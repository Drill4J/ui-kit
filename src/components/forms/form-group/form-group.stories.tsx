import { ComponentMeta, Story } from '@storybook/react';

import React from 'react';
import { FormGroup } from './form-group';
import { Input } from '../inputs/input';
import { Textarea } from '../inputs/textarea';
import { Icons } from '../../icon';

export default {
  title: 'FormGroup',
  component: FormGroup,
} as ComponentMeta<typeof FormGroup>;

const TemplateWithInput: Story = (args) => (
  <div tw="w-[300px]">
    <FormGroup label="" {...args}>
      <Input />
    </FormGroup>
  </div>
);

const TemplateWithTextarea: Story = (args) => (
  <div tw="w-[300px]">
    <FormGroup label="" {...args}>
      <Textarea />
    </FormGroup>
  </div>
);

export const DefaultWithInput = TemplateWithInput.bind({});
DefaultWithInput.args = { label: 'default', optional: false };

export const OptionalWithInput = TemplateWithInput.bind({});
OptionalWithInput.args = { label: 'default', optional: true };

export const OptionalWithInputAndAction = TemplateWithInput.bind({});
OptionalWithInputAndAction.args = { label: 'default', optional: true, actions: <Icons.Info /> };

export const DefaultWithTextarea = TemplateWithTextarea.bind({});
DefaultWithTextarea.args = { label: 'default', optional: false };

export const OptionalWithTextarea = TemplateWithTextarea.bind({});
OptionalWithTextarea.args = { label: 'default', optional: true };

export const OptionalWithTextareaAndAction = TemplateWithTextarea.bind({});
OptionalWithTextareaAndAction.args = { label: 'default', optional: true, actions: <Icons.Info /> };
