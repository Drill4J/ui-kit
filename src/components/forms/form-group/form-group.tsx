import tw, { styled } from 'twin.macro';

interface Props {
  children?: React.ReactNode;
  label?: string;
  optional?: boolean;
  actions?: React.ReactNode;
}

export const FormGroup = ({
  children, label, optional, actions, ...rest
}: Props) => (
  <div {...rest}>
    <Panel>
      <div tw="font-bold text-14 leading-20 text-monochrome-black">{label}</div>
      <div>
        <Panel>
          {optional && <div tw="text-12 leading-16 text-monochrome-default">Optional</div>}
          {actions && <div tw="flex ml-2">{actions}</div>}
        </Panel>
      </div>
    </Panel>
    <div tw="w-full mt-2">{children}</div>
  </div>
);

const Panel = styled.div`
  ${tw`flex w-full items-center justify-between`};
`;
