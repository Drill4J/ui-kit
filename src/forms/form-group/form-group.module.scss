@import '../../style-guide/index';

.form-group {
  &__label {
    font-family: $semibold-font;
    font-size: 14px;
    line-height: 40px;
    color: map-get($monochrome, black);
  }

  &__optional-label {
    font-size: 12px;
    color: map-get($monochrome, default);
  }

  &__actions {
    display: flex;
    margin-left: 8px;
  }

  &__input {
    width: 100%;
  }
}
