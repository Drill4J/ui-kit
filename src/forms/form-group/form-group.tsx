import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Panel } from '../../layouts';

import styles from './form-group.module.scss';

interface Props {
  className?: string;
  children?: React.ReactNode;
  label?: string;
  optional?: boolean;
  actions?: React.ReactNode;
}

const formGroup = BEM(styles);

export const FormGroup = formGroup(({
  className, children, label, optional, actions,
}: Props) => (
  <div className={className}>
    <Panel align="space-between">
      <Label>{label}</Label>
      <div>
        <Panel>
          {optional && <OptionalLabel>Optional</OptionalLabel>}
          {actions && <Actions>{actions}</Actions>}
        </Panel>
      </div>
    </Panel>
    <Input>{children}</Input>
  </div>
));

const Label = formGroup.label('label');
const OptionalLabel = formGroup.optionalLabel('span');
const Actions = formGroup.actions('div');
const Input = formGroup.input('div');
