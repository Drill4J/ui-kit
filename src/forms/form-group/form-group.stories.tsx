import { storiesOf } from '@storybook/react';

import { FormGroup } from './form-group';
import { Input } from '../inputs/input';
import { Icons } from '../../components/icon';

storiesOf('FormGroup with all props', module).add('FormGroup', () => (
  <FormGroup
    label="Form"
    optional
    actions={<Icons.Info />}
  >
    <Input />
  </FormGroup>
))
  .add('FormGroup with optional prop', () => (
    <FormGroup
      label="Form"
      optional
    >
      <Input />
    </FormGroup>
  ))
  .add('FormGroup with actions prop', () => (
    <FormGroup
      label="Form"
      actions={<Icons.Info />}
    >
      <Input />
    </FormGroup>
  ));
