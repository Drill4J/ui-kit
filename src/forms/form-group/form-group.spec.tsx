import renderer from 'react-test-renderer';

import { FormGroup } from './form-group';

it('should match snapshot', () => {
  const tree = renderer
    .create(
      <FormGroup
        label="Form"
        optional
      >
        <input />
      </FormGroup>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
