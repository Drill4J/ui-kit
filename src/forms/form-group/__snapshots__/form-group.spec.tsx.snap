// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    className="sc-bdvvaa bkKWSB"
  >
    <label
      className="sc-gsDJrp gufwci"
    >
      Form
    </label>
    <div>
      <div
        className="sc-bdvvaa bkKWSB"
      >
        <span
          className="sc-dkPtyc iUmdcE"
        >
          Optional
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-eCImvq cAEScb"
  >
    <input />
  </div>
</div>
`;
